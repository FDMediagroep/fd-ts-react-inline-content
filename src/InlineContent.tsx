import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
