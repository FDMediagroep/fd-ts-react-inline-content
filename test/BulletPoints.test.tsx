import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BulletPoints } from '../src/InlineContent';

describe('BulletPoints', ()  => {
    const bullets = [
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
        'Vivamus varius sed quam vel aliquam.',
        'Ut eleifend porta arcu, a eleifend enim viverra sit amet.',
        'Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.'
    ];

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<BulletPoints
            bullets={bullets}
        />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<BulletPoints
            bullets={bullets}
            align='right'
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<BulletPoints
            bullets={bullets}
            align='block'
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
