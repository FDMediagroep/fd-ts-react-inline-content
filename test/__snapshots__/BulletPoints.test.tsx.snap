// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPoints should render align block correctly 1`] = `
<div
  className="inline-content bullet-points block"
>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Vivamus varius sed quam vel aliquam.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;

exports[`BulletPoints should render align right correctly 1`] = `
<div
  className="inline-content bullet-points right"
>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Vivamus varius sed quam vel aliquam.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;

exports[`BulletPoints should render align right correctly 2`] = `
<div
  className="inline-content bullet-points right"
>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Vivamus varius sed quam vel aliquam.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Ut eleifend porta arcu, a eleifend enim viverra sit amet.
  </p>
  <p
    className="fd-typography article-p-inline-bullet-points"
    textStyle="article-p-inline-bullet-points"
  >
    Quisque a commodo eros. Sed at mi consequat, lacinia eros sed, luctus velit.
  </p>
</div>
`;
